### PR TITLE
feat: compressed token program - registerSpl, approveAndMintTo 

### PR DESCRIPTION
**Pull request by SwenSchaeferjohann**
* support for existing SPL mints. 
* we need a dedicated method for these because existing mints have external (relative to our system) mint authorities, therefore our system cannot directly mintTo compressed accounts. 

- [x] typedoc and cleanup compressed token program
- [x] instructions
- [x] actions
- [x] tests
- [x] cli commands
- [x] cli tests

### CODE REVIEW

File: cli/src/commands/approve-and-mint-to/index.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      required: true,
+    }),
+    to: Flags.string({
+      description: "Specify the recipient address.",
+      required: true,
+    }),
+    amount: Flags.integer({
+      description: "Amount to mint, in tokens.",
+      required: true,
+    }),
```
Comment (ananas-block) [inline cli/src/commands/approve-and-mint-to/index.ts:38]: for required flags wouldn't it make sense to make these arguments?

File: js/compressed-token/src/types.ts [hunk 2]
```diff
@@ -2,54 +2,115 @@ import { PublicKey } from '@solana/web3.js';
...
     lamports: BN | null;
 };
 
 export type InputTokenDataWithContext = {
+    /**
+     * The amount of tokens to transfer
+     */
     amount: BN;
+    /**
+     * Optional: The index of the delegate in the merkle tree
```
Comment (ananas-block) [inline js/compressed-token/src/types.ts:26]: ``delegate in the merkle tree`` what does this mean?

Comment (ananas-block) [reply js/compressed-token/src/types.ts:26]: is it the index in remaining accounts of the compressed tx?

**Review by ananas-block — APPROVED** [review]
looks good overall, no need to swap any flags for args right away but it feels more intuitive to me.
Also see my other comment.

File: cli/src/commands/approve-and-mint-to/index.ts (hunk 1 above)
Comment (SwenSchaeferjohann) [reply cli/src/commands/approve-and-mint-to/index.ts:38]: on bucket list for separate pr

**Pull request merged**